main/relational_to_grid.es6.js

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+const path = require('path');
+const process = require('process');
+const url = require('url');
+
+require('foam2');
+
+require('../lib/grid_dao.es6.js');
+require('../lib/http_json_dao.es6.js');
+require('../lib/json_dao_container.es6.js');
+require('../lib/local_json_dao.es6.js');
+require('../lib/web_apis/release.es6.js');
+require('../lib/web_apis/release_interface_relationship.es6.js');
+require('../lib/web_apis/web_interface.es6.js');
+const pkg = org.chromium.apis.web;
+
+const logger = foam.log.ConsoleLogger.create();
+
+const USAGE = `USAGE:
+
+    node /path/to/import.es6.js BaseConfluenceDataURL
+
+        BaseConfluenceDataURL = absolute https: or file: URL to directory
+                                where Confluence data are stored with NO
+                                trailing slash.`;
+if (process.argv.length !== 3) {
+  console.error(USAGE);
+  process.exit(1);
+}
+
+const dataUrl = url.parse(process.argv[2]);
+if (dataUrl.protocol !== 'file:' && dataUrl.protocol !== 'https:') {
+  console.error('BaseConfluenceDataURL parameter must be file: or https: URL');
+}
+
+let container = pkg.JsonDAOContainer.create({
+  mode: dataUrl.protocol === 'file:' ? pkg.JsonDAOContainerMode.LOCAL :
+      pkg.JsonDAOContainerMode.HTTP,
+  basename: dataUrl.protocol === 'file:' ? dataUrl.pathname : url.format(dataUrl),
+}, logger);
+
+logger.info('Gathering Confluence data from JSON');
+
+Promise.all([
+  container.releaseDAO.orderBy(pkg.Release.RELEASE_DATE).select(),
+  container.webInterfaceDAO.orderBy(pkg.WebInterface.ID).select(),
+  container.releaseWebInterfaceJunctionDAO.select(),
+]).then(function(arraySinks) {
+  logger.info('JSON data gathered');
+
+  const releases = arraySinks[0].array;
+  const apis = arraySinks[1].array;
+  const joins = arraySinks[2].array;
+
+  const outputter = foam.json.Outputter.create({
+    pretty: false,
+    formatDatesAsNumbers: true,
+    outputDefaultValues: false,
+    useShortNames: false,
+    strict: true,
+  });
+  function store(arraySink) {
```

Comment:
Maybe this is just a promise-ism/JSism, but this looks like it could have been defined outside of the promise, and made following along with the logic of the promise a little easier.

Reply:
I think I follow what you mean. I moved `outputter` and `store` to the global scope/context.